### DISCUSSION TITLE
ModuleNotFoundError: No module named 'tzlocal'

### DISCUSSION THREAD

**Discussion by snweaus** [mcguirepr89/BirdNET-Pi#613] [General]
Hi all, 

I'm hoping someone can help. I've just got BirdNET-Pi up and running for the first time on a Raspberry Pi 3B+ 

I can access the server on my local network however after playing a number of magpie videos have not had any detections. When I check the log it is repeatedly throwing an error "No module named 'tzlocal'" 

See screenshot below.

<img width="638" alt="Screen Shot 2022-10-20 at 9 40 51 pm" src="https://user-images.githubusercontent.com/116271733/196928059-460f30d9-4477-496a-8071-d3ed71dc2f27.png">

I am accessing the server via Chrome. 

I've tried (re)installing the module "tzlocal" using Pip with:

- sudo pip3 install tzlocal
- sudo pip install tzlocal

Then rebooted and no change to the issue.

Any help would be greatly appreciated.

**Comment by ehpersonal38** [comment]
Hi @snweaus ! 

BirdNET-Pi uses Python under a virtual environment, so you'll have to execute the following command to enter that environment and install the required packages. This should've been done during installation but seemed to have failed on your system for some reason?

```
source /home/pi/BirdNET-Pi/birdnet/bin/activate &&
pip3 install -U -r /home/pi/BirdNET-Pi/requirements.txt &&
deactivate
```

Then after a reboot or `bash ~/BirdNET-Pi/scripts/restart_services.sh`, hopefully the issue will be fixed!

**Reply by snweaus** [reply]
Thanks so much, @ehpersonal38 !

That did the trick :)

**Reply by NilsEinsfelder** [reply]
Nearly worked for me as well: The "tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl" wasn't found. Since the file is already on the Raspi,I could manually pip the file from the BirdNET-Pi directory. Afterwards i took it out of the requirements.txt to avoid any mess. That did the jobs, thank you very much! :-)

**Reply by GreenOrpheus** [reply]
> Nearly worked for me as well: The "tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl" wasn't found. Since the file is already on the Raspi,I could manually pip the file from the BirdNET-Pi directory. Afterwards i took it out of the requirements.txt to avoid any mess. That did the jobs, thank you very much! :-)

@NilsEinsfelder, could you explain in layman's terms what you did? I've tried the above fix by @ehpersonal38, but it just results in a "-bash: /home/pi/BirdNET-Pi/birdnet/bin/activate: No such file or directory" error.

**Reply by GreenOrpheus** [reply]
Ok, so I tried asking chatgpt for advice on this problem. Lol. It was very quick to suggest some things, and I did end up learning a little bit about how to navigate around in powershell, but ultimately our AI overlord's advice corrupted my install. I did a fresh install (for the 5th time) and everything worked. No errors, just bird noises.

The only thing I can think of was my internet is not the greatest, and the installation must have been missing something due to disconnects/reconnects.
